v2/index/indexsorted.go

```diff
@@ -70,16 +69,12 @@ func (s *singleWidthIndex) Unmarshal(r io.Reader) error {
 	return err
 }
 
-func (s *singleWidthIndex) Less(i int, digest []byte) bool {
-	return bytes.Compare(digest[:], s.index[i*int(s.width):((i+1)*int(s.width)-8)]) <= 0
+func (s *singleWidthIndex) Less(i int, hash []byte) bool {
+	return bytes.Compare(hash[:], s.index[i*int(s.width):((i+1)*int(s.width)-8)]) <= 0
 }
 
 func (s *singleWidthIndex) Get(c cid.Cid) (uint64, error) {
-	d, err := multihash.Decode(c.Hash())
-	if err != nil {
-		return 0, err
-	}
-	return s.get(d.Digest), nil
```

Comment:
this is an on-disk change to the format of the index / needs to be reflected in spec?

I think the point here was that it'll be the same byte indicating the type of multihash on every entry, so why are we repeating that every time?

internally it seems like it should be okay to store just the digest, no?

Reply:
You're right, this breaks away from the spec.

Technically I could construct two multihashes with the same digest but different multihash codes; the blockstore compares the whole multihash so it would treat them as different, but this index only looks at the digest length and digest bytes, so it would treat them as equal.

But in practice, only storing the digest in each digest-length bucket saves space, and is going to be enough for uniqueness for basically every user. It should only cause problems if someone ends up with the same digest on different hashing algorithms, which is extremely unlikely.

I'll update the PR to revert the changes to the index.